core/src/main/scala/kafka/log/Log.scala

```diff
@@ -157,7 +169,7 @@ class Log(@volatile var dir: File,
     // The earliest leader epoch may not be flushed during a hard failure. Recover it here.
     leaderEpochCache.clearEarliest(logStartOffset)
 
-    buildAndRecoverPidMap(logEndOffset)
+    loadProducerState(logEndOffset)
```

Comment:
Do we need to do this under lock since we are updating the state in ProducerStateManager?